sw/device/silicon_creator/lib/boot_svc/boot_svc_ownership_activate.h

```diff
@@ -0,0 +1,113 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_BOOT_SVC_BOOT_SVC_OWNERSHIP_ACTIVATE_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_BOOT_SVC_BOOT_SVC_OWNERSHIP_ACTIVATE_H_
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/base/chip.h"
+#include "sw/device/silicon_creator/lib/boot_svc/boot_svc_header.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/nonce.h"
+#include "sw/device/silicon_creator/lib/ownership/datatypes.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+enum {
+  // ASCII: AORQ.
+  kBootSvcOwnershipActivateReqType = 0x51524f41,
+  // ASCII: AORS.
+  kBootSvcOwnershipActivateResType = 0x53524f41,
+};
+
+/**
+ * A Next Boot BL0 Slot request message.
+ */
+typedef struct boot_svc_ownership_activate_req {
+  /**
+   * Boot services message header.
+   */
+  boot_svc_header_t header;
+  /**
+   * Unlock mode: Any, Endorsed, Update or Abort.
+   */
+  uint32_t primary_bl0_slot;
+  /**
+   * Erase previous owner's flash (hardened_bool_t).
+   */
+  uint32_t erase_previous;
+  /**
+   * Reserved for future use.
+   */
+  uint32_t reserved[33];
+  /**
+   * The current ownership nonce.
+   */
+  nonce_t nonce;
+  /**
+   * Signature over [unlock_mode..next_owner_key]
+   */
+  owner_signature_t signature;
+
+} boot_svc_ownership_activate_req_t;
+
+OT_ASSERT_MEMBER_OFFSET(boot_svc_ownership_activate_req_t, header, 0);
+OT_ASSERT_MEMBER_OFFSET(boot_svc_ownership_activate_req_t, primary_bl0_slot,
+                        CHIP_BOOT_SVC_MSG_HEADER_SIZE);
+OT_ASSERT_MEMBER_OFFSET(boot_svc_ownership_activate_req_t, erase_previous,
+                        CHIP_BOOT_SVC_MSG_HEADER_SIZE + 4);
+OT_ASSERT_MEMBER_OFFSET(boot_svc_ownership_activate_req_t, reserved,
+                        CHIP_BOOT_SVC_MSG_HEADER_SIZE + 8);
+OT_ASSERT_MEMBER_OFFSET(boot_svc_ownership_activate_req_t, nonce,
+                        CHIP_BOOT_SVC_MSG_HEADER_SIZE + 140);
+OT_ASSERT_MEMBER_OFFSET(boot_svc_ownership_activate_req_t, signature,
+                        CHIP_BOOT_SVC_MSG_HEADER_SIZE + 148);
+OT_ASSERT_SIZE(boot_svc_ownership_activate_req_t, 256);
+
+/**
+ * A Next Boot BL0 Slot response message.
+ */
+typedef struct boot_svc_ownership_activate_res {
```

Comment:
This is something that just occurred to me and that doesn't need to block this PR. Do you think it may be worth it for us to evaluate if any of the responses from the device need to be signed? This would imply that we need the `nonce` to also be part of the response along with the signature.

Reply:
I hadn't considered signed responses.  A boot-services attestation response is one such example, but in that example, I think we'd want to sign a user-supplied nonce rather than a silicon_creator controlled nonce.